src/controllers/Controller.js

```diff
@@ -0,0 +1,112 @@
+import { Console } from "@woowacourse/mission-utils";
+
+import Opponent from "../models/Opponent.js";
+import User from "../models/User.js";
+
+import GAME_START_VIEW from "../views/GameStartView.js";
+import INPUT_NUMBER_VIEW from "../views/InputNumberView.js";
+import OUTPUT_VIEW from "../views/OutputView.js";
+
+import RANDOM_NUMBER from "../utils/RandomNumber.js";
+import CHECK_VALIDATION from "../utils/CheckValidation.js";
+
+import MESSAGES from "../constants/Messages.js";
+import CONDITIONS from "../constants/Conditions.js";
+import RESTART_VIEW from "../views/RestartView.js";
+
+/**
+ * @class Controller
+ * 1. 상대방(컴퓨터) 숫자를 생성한다
+ * 2. 상대방의 숫자와 사용자가 입력한 숫자를 비교하여 평가한다
+ */
+class Controller {
+  /** @type {Opponent} */
+  #opponent;
+
+  /** @type {User} */
+  #user;
+
+  /**
+   * 게임을 실행하는 함수
+   * @returns {boolean} 재시작 여부
+   */
+  async run() {
+    // 게임 시작
+    GAME_START_VIEW();
+    this.generateOpponent();
+
+    // 숫자 입력 후 평가
+    this.#user = new User();
+    await this.matchNumber();
+
+    // 재시작 여부 입력
+    await this.saveUserRestart();
+    return this.#user.getRestart();
+  }
+
+  /**
+   * 상대방(컴퓨터) 숫자를 생성하는 함수
+   * @returns {Opponent}
+   */
+  generateOpponent() {
+    this.#opponent = new Opponent(RANDOM_NUMBER());
+    return this.#opponent;
+  }
+
+  /**
+   * 상대방(컴퓨터)의 숫자와 사용자의 숫자를 비교하여 평가하는 함수
+   */
+  async matchNumber() {
+    const opponentNumberElements = this.#opponent
+      .getNumber()
+      .toString()
+      .split("")
+      .map(Number);
+
+    let strikeCount = 0;
+    let ballCount = 0;
```

Comment:
미리 초기화하지 않아도 잘 동작합니다! 

```suggestion
    let strikeCount;
    let ballCount;
```

Reply:
저는 개인적으로 count의 초기값을 정해준 모습이 좋다고 봅니다!

Reply:
저도 초기화 하는 게 좀 더 좋을 것 같아서 초기화해봤습니다!